go/test/endtoend/onlineddl/flow/onlineddl_flow_test.go

```diff
@@ -0,0 +1,664 @@
+				// This workload will consider throttling state and avoid generating DMLs if throttled.
+				wg.Add(1)
+				go func() {
+					defer cancel()
+					defer t.Logf("Terminating workload")
+					defer wg.Done()
+					runMultipleConnections(workloadCtx, t)
+				}()
+			})
+			appliedDMLStart := totalAppliedDML.Load()
+
+			hint := "post_completion_hint"
+			var uuid string
+			t.Run("submit migration", func(t *testing.T) {
+				uuid = testOnlineDDLStatement(t, fmt.Sprintf(alterHintStatement, hint), "online --postpone-completion", "", true)
+			})
+			t.Run("wait for ready_to_complete", func(t *testing.T) {
+				waitForReadyToComplete(t, uuid, true)
+			})
+			t.Run("validating running status", func(t *testing.T) {
+				onlineddl.CheckMigrationStatus(t, &vtParams, shards, uuid, schema.OnlineDDLStatusRunning)
+			})
+			t.Run("throttle online-ddl", func(t *testing.T) {
+				onlineddl.ThrottleAllMigrations(t, &vtParams)
+				onlineddl.CheckThrottledApps(t, &vtParams, throttlerapp.OnlineDDLName, true)
+
+				for _, tab := range tablets {
+					body, err := throttleApp(tab.VttabletProcess, throttlerapp.OnlineDDLName)
+					assert.NoError(t, err)
+					assert.Contains(t, body, throttlerapp.OnlineDDLName)
+				}
+				waitForThrottleCheckStatus(t, throttlerapp.OnlineDDLName, primaryTablet, http.StatusExpectationFailed)
+			})
+			t.Run("unthrottle online-ddl", func(t *testing.T) {
+				onlineddl.UnthrottleAllMigrations(t, &vtParams)
+				onlineddl.CheckThrottledApps(t, &vtParams, throttlerapp.OnlineDDLName, false)
+
+				for _, tab := range tablets {
+					body, err := unthrottleApp(tab.VttabletProcess, throttlerapp.OnlineDDLName)
+					assert.NoError(t, err)
+					assert.Contains(t, body, throttlerapp.OnlineDDLName)
+				}
+				waitForThrottleCheckStatus(t, throttlerapp.OnlineDDLName, primaryTablet, http.StatusOK)
+			})
+			t.Run("additional wait", func(t *testing.T) {
+				select {
+				case <-time.After(3 * time.Second):
+				case <-ctx.Done():
+					require.Fail(t, "context cancelled")
+				}
+			})
+			t.Run("validate applied DML", func(t *testing.T) {
+				// Validate that during Online DDL, and even with throttling, we were
+				// able to produce meaningful traffic.
+				appliedDMLEnd := totalAppliedDML.Load()
+				assert.Greater(t, appliedDMLEnd, appliedDMLStart)
+				assert.GreaterOrEqual(t, appliedDMLEnd-appliedDMLStart, int64(maxTableRows))
+				t.Logf("Applied DML: %d", appliedDMLEnd-appliedDMLStart)
+			})
+			t.Run("attempt to complete", func(t *testing.T) {
+				onlineddl.CheckCompleteMigration(t, &vtParams, shards, uuid, true)
+			})
+			isComplete := false
+			t.Run("optimistic wait for migration completion", func(t *testing.T) {
+				select {
+				case <-time.After(10 * time.Second):
+				case <-ctx.Done():
+					require.Fail(t, "context cancelled")
+				}
+
+				status := onlineddl.WaitForMigrationStatus(t, &vtParams, shards, uuid, migrationWaitTimeout, schema.OnlineDDLStatusRunning, schema.OnlineDDLStatusComplete)
```

Comment:
I'm curious why in this spot and similar ones we're not instead just adding 10 seconds to the `migrationWaitTimeout` we pass in?

Reply:
Huh. You're right and this is not needed. Removing altogether.

Reply:
fixed